### PR TITLE
Add openpubkey blog post.

### PR DESCRIPTION
**Pull request by dlorenc**

### CODE REVIEW

**Comment by dlorenc** [comment]
Thanks for the thorough reviews @kommendorkapten and @haydentherapper

**Review by kommendorkapten — APPROVED** [review]

**Comment by lukehinds** [comment]
Looks good to me thanks Dan. The only suggestion I would make is to do something like we did for the spicy blockchain / sigstore post, with an author / community disclaimer: https://blog.sigstore.dev/sigstore-blockchain-vs-transparency-logs-d673ea41a9be/ 

This is useful for if someone in the community disagrees / feels misrepresented, we can state its the authors views and not necessarily the whole community.

**Review by lukehinds — APPROVED** [review]

**Comment by lukehinds** [comment]
> > Looks good to me thanks Dan. The only suggestion I would make is to do something like we did for the spicy blockchain / sigstore post, with an author / community disclaimer: https://blog.sigstore.dev/sigstore-blockchain-vs-transparency-logs-d673ea41a9be/
> 
> Thanks, added!

Cool, TSC are aware and should review shortly, I will keep a window open for them, but realise we want to respond while the topic is fresh.

File: content/openpubkey-and-sigstore.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+There’s a field you can include whatever you want in that will be signed by someone you trust in the response!
+If we embed a public key and an identity in there, we can squint and call that a certificate! We can even include some random data to preserve the replay-resistance property of the protocol. 
+
+# Tradeoff #1 - Server-Side vs. Client-Side Complexity
+This is the first tradeoff, OpenPubKey eliminates some server side complexity - no one needs to run a certificate authority (Fulcio), but clients need to perpetually track all the signing keys used by all identity providers to verify historical signatures.
+If you start tracking them today, you can’t verify signatures generated earlier than today.
+If you miss one because your system goes down, you might not be able to verify an authentic signature.
+
+There are ways around this - someone could run a central log of public keys over time.
+The current OpenPubKey reference implementation contains a TODO with just that - fetch the keys from some log somewhere.
```
Comment (imjasonh) [inline content/openpubkey-and-sigstore.md:169]:
```suggestion
The current OpenPubKey reference implementation [contains a `TODO` with just that](https://github.com/openpubkey/openpubkey/blob/11849ba7f67611180997db3676905dc26da13b20/parties/githubclient.go#L172) - fetch the keys from some log somewhere.
```

**Review by cpanato — DISMISSED** [review]

**Review by trevrosen — APPROVED** [review]

**Review by priyawadhwa — DISMISSED** [review]

**Review by bobcallaway — APPROVED** [review]

File: content/openpubkey-and-sigstore.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+This explanation is a bit over-simplified.
+OpenPubKey actually uses two tokens (the original ID token which contains the nonce, and their own PK token which effectively wraps the ID token and contains the public key.
+The full setup looks like this:
+
+![](/images/openpubkey.png)
+
+OpenPubKey also contains support for expiration - the default in this mode is for these certificates to be valid for two weeks.
+The paper doesn’t really specify what happens after that, or how to handle revocation, but we can imagine ways these can be constructed on top of this protocol.
+
+This effectively eliminates the Fulcio component by using the token directly as a certificate.
```
Comment (SantiagoTorres) [inline content/openpubkey-and-sigstore.md:90]: I'd rephrase this as "this effectively removes the OIDC part of Fulcio (but not its transparency log)" or so

File: content/openpubkey-and-sigstore.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+The OpenPubKey implementation (not the paper) has an [issue and a potential workaround](https://github.com/openpubkey/openpubkey/issues/7) to this by using zero-knowledge proofs - it looks promising but
+I’m not enough of an expert to know if it will work out.
+
+# Wrapping Up
+
+I believe the OpenPubKey approach is clever and might solve some use-cases that Sigstore doesn’t solve well.
+I’m admittedly biased, having participated deeply in the design and implementation of Sigstore, but I believe these tradeoffs (the potential privacy leak of information contained in the identity Tokens, and the large amount of complexity for expired key management pushed to all clients) favor Sigstore for most supply chain security use-cases.
+
+The added complexity of using a central certificate authority has been mostly mitigated by the excellent efforts of the Sigstore community and sponsoring companies, including Stacklok, Chainguard, Google, Red Hat, and GitHub.
```
Comment (SantiagoTorres) [inline content/openpubkey-and-sigstore.md:209]: Can we add "and academic partners like Purdue University" here :) ?

**Review by SantiagoTorres — APPROVED** [review]
I'm throwing it two nits. I like the way it's presented. I believe that this perhaps is not suitable for long lived things (i.e., supply chain), but perhaps it was never meant to?

**Review by priyawadhwa — APPROVED** [review]

**Comment by priyawadhwa** [comment]
cc @trevrosen @bobcallaway @SantiagoTorres @lukehinds I think we need 2 more approvals for this to merge

**Review by trevrosen — APPROVED** [review]

**Comment by joshuagl** [comment]
> I'm throwing it two nits. I like the way it's presented. I believe that this perhaps is not suitable for long lived things (i.e., supply chain), but perhaps it was never meant to? 

There's an in-toto attestation signer implementation in their GH org, which suggests they are targeting supply chain use-casea

**Review by SantiagoTorres — APPROVED** [review]
